File: src/main/scala/nu/tengstrand/tetrisanalyzer/piece/PieceO.scala
package nu.tengstrand.tetrisanalyzer.piece

object PieceO {
  def apply() = new PieceO
}

class PieceO extends Piece {
  val number = 7.toByte
  val character = 'O'
  protected val widths = Array(2)
  protected val heights = Array(2)
  protected val shapes = Array(
    new PieceShape(Array(Point(0,0), Point(1,0), Point(0,1), Point(1,1)))
  )
}

File: langs/java/src/main/java/com/github/tetrisanalyzer/piece/PieceO.java
package com.github.tetrisanalyzer.piece;

import com.github.tetrisanalyzer.settings.PieceSettings;

public class PieceO extends Piece {

    public PieceO(PieceSettings settings) {
        super(settings);
    }

    @Override public byte number() { return 7; }
    @Override public char character() { return 'O'; }
    @Override protected int[] widths() { return new int[] { 2 }; }
    @Override protected int[] heights() { return new int[] { 2 }; }

    @Override protected PieceShape[] shapes() {
        return new PieceShape[] {
            new PieceShape(new Point(0,0), new Point(1,0), new Point(0,1), new Point(1,1))
        };
    }
}
